src/vm/comutilnative.cpp

```diff
@@ -1295,10 +1295,25 @@ FCIMPL1(INT64, GCInterface::GetTotalAllocatedBytes, CLR_BOOL precise)
     {
         // NOTE: we do not want to make imprecise flavor too slow. 
         // As it could be noticed we read 64bit values that may be concurrently updated.
-        // Such reads are not guaranteed to be atomic on 32bit and inrare cases we may see torn values resultng in outlier results.
+        // Such reads are not guaranteed to be atomic on 32bit and in rare cases we may see torn values resultng in outlier results.
         // That would be extremely rare and in a context of imprecise helper is not worth additional synchronization.
```

Comment:
It is rare, but when it happens - we may see the value to jump by 4GB and then not change for a while until the allocations catch up.

Reply:
This value is a performance counter. Performance counters are often used for anomaly monitoring where even rare flukes are a problem. 

I am not sure about the correctness vs. performance tradeoff for this API. I think it may be better to  make the API reliable without flukes even if it means that it is a bit slower.

Reply:
Yes, I thought about that, but could not decide. Last time we decided it is ok since the chances are extremely low and only on 32 bit and it would be just a quick blip.
Now the blip will last longer.

I will make it an atomic read on 32bit.